controllers/agentconfig_controller.go

```diff
@@ -588,10 +584,9 @@ func (r *AgentConfigReconciler) isReadyToBeDeleted(ctx context.Context, log logr
 func (r *AgentConfigReconciler) getExistingPluginPVCs(ctx context.Context, log logr.Logger, agentCfg *porterv1.AgentConfigAdapter) (readyPVC *corev1.PersistentVolumeClaim, tempPVC *corev1.PersistentVolumeClaim, err error) {
 	results := &corev1.PersistentVolumeClaimList{}
 	err = r.List(ctx, results, client.InNamespace(agentCfg.Namespace), client.MatchingLabels(agentCfg.Spec.Plugins.GetLabels()))
-	if err != nil && !apierrors.IsNotFound(err) {
+	if err != nil && !apierrors.IsNotFound(err) || len(results.Items) < 1 {
```

Comment:
If this is < 1, this error could be nil as the List can return without failing.  _(This matches the behavior if there are no PVCs found and fallthrough line 612)_